sui-execution/latest/sui-adapter/src/execution_engine.rs

```diff
@@ -220,100 +218,69 @@ fn execute_transaction<Mode: ExecutionMode>(
         gas_charger.no_charges(),
         "No gas charges must be applied yet"
     );
-    let is_genesis_tx = matches!(transaction_kind, TransactionKind::Genesis(_));
-    let advance_epoch_gas_summary = transaction_kind.get_advance_epoch_tx_gas_summary();
 
     // We must charge object read here during transaction execution, because if this fails
     // we must still ensure an effect is committed and all objects versions incremented
     let result = gas_charger.charge_input_objects(temporary_store);
     let mut result = result.and_then(|()| {
-        let mut execution_result = if deny_cert {
-            Err(ExecutionError::new(
-                ExecutionErrorKind::CertificateDenied,
-                None,
-            ))
-        } else {
-            execution_loop::<Mode>(
-                temporary_store,
-                transaction_kind,
-                tx_ctx,
-                move_vm,
-                gas_charger,
-                protocol_config,
-                metrics.clone(),
-            )
-        };
+        let execution_result = execution_loop::<Mode>(
+            temporary_store,
+            transaction_kind.clone(),
+            tx_ctx,
+            move_vm,
+            gas_charger,
+            protocol_config,
+            metrics.clone(),
+        );
 
-        let effects_estimated_size = temporary_store.estimate_effects_size_upperbound();
+        check_meter_limit(
```

Comment:
I do not think we want to return here, right?
we want to assign to execution result or pass that in, is that correct?

Reply:
No tests failed, but I have reverted it to not return until the end 👍